dianatanase/assignment2/Q1_tests.java

```diff
@@ -0,0 +1,124 @@
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import org.junit.Test;
+
+
+public class Q1_tests {
+
+	// Test result is an empty array list for an empty or null tree.
+	@Test
+	public void testNull(){
+		
+		Tree<Integer> tree = new Tree<Integer>();
+		assertEquals(0, Q1.getAncestorsInTree(5, tree).size());
+		
+		assertEquals(0, Q1.getAncestorsInTree(5, null).size());
+	}
+	
+	// Test the result with keys equal with data and in increasing order.
+	@Test
+	public void testKeysInOrder() {
+		
+		// Initialize nodes of the tree.
+		Node<Integer> node1 = new Node<Integer>(null, 1, 1, null, null);
```

Comment:
How else could we construct this graph?
Node<Integer> tree = new Tree<Integer>();
tree.SetRoot(1);
tree.AddChildToIndex(1, 2);
tree.AddChildToIndex(1, 3);
tree.AddChildToIndex(2, 4);
tree.AddChildToIndex(2, 5);
tree.AddChildToIndex(3, 6);
tree.AddChildToIndex(3, 7);
and so on. Which one is more readable?